tests/resources/event_gateway_reusable_test.go

```diff
@@ -0,0 +1,2021 @@
+			hclbuilder.KonnectBeta,
+			fmt.Sprintf(providerConfigTemplate, serverScheme, serverHost, serverPort),
+		)
+		builder.ProviderProperty = hclbuilder.KonnectBeta
+
+		egwCp, err := hclbuilder.FromString(`
+			resource "konnect_event_gateway" "tf-test-egw-decrypt" {
+			  name = "tf-test-egw-decrypt"
+			}
+		`)
+		require.NoError(t, err)
+
+		backend, err := hclbuilder.FromString(`
+			resource "konnect_event_gateway_backend_cluster" "tf-test-backend" {
+			  name = "tf-test-backend"
+
+			  authentication = {
+				anonymous = {}
+			  }
+
+			  bootstrap_servers = ["127.0.0.1:9092"]
+
+			  tls = {
+				enabled = false
+				insecure_skip_verify = true
+			  }
+			}
+		`)
+		require.NoError(t, err)
+
+		backend.AddAttribute(
+			"gateway_id",
+			egwCp.ResourcePath()+".id",
+		)
+
+		virtualCluster, err := hclbuilder.FromString(`
+			resource "konnect_event_gateway_virtual_cluster" "tf-test-virtual" {
+			  name      = "tf-test-virtual"
+			  dns_label = "tf-test-vc"
+			  acl_mode  = "passthrough"
+
+			  authentication = [
+				{
+				  anonymous = {}
+				}
+			  ]
+
+			  destination = {
+				id = konnect_event_gateway_backend_cluster.tf-test-backend.id
+			  }
+			}
+		`)
+		require.NoError(t, err)
+
+		virtualCluster.AddAttribute(
+			"gateway_id",
+			egwCp.ResourcePath()+".id",
+		)
+
+		staticKey, err := hclbuilder.FromString(`
+			resource "konnect_event_gateway_static_key" "tf-test-static-key" {
+			  name  = "tf-test-static-key"
+			  value = "$${vault.env[\"MY_ENV_VAR\"]}"
+			}
+		`)
+		require.NoError(t, err)
+
+		staticKey.AddAttribute(
+			"gateway_id",
+			egwCp.ResourcePath()+".id",
+		)
+
+		decryptPolicy, err := hclbuilder.FromString(`
+			resource "konnect_event_gateway_consume_policy_decrypt" "tf-test-decrypt-policy" {
+			  name        = "tf-test-decrypt-policy"
+			  description = "initial decrypt policy"
+			  enabled     = false
+			  condition   = "context.topic.name.endsWith('my_suffix')"
+
+			  config = {
+				failure_mode = "passthrough"
+
+				key_sources = [
+				  {
+					static = {
+					  name = "tf-test-static-key"
+					}
+				  }
+				]
+				part_of_record = ["key"]
+			  }
+			}
+		`)
+		require.NoError(t, err)
+
+		decryptPolicy.AddAttribute(
+			"gateway_id",
+			egwCp.ResourcePath()+".id",
+		)
+		decryptPolicy.AddAttribute(
+			"virtual_cluster_id",
+			virtualCluster.ResourcePath()+".id",
+		)
+		resource.Test(t, resource.TestCase{
+			ProtoV6ProviderFactories: providerFactory,
+			Steps: []resource.TestStep{
+				{
+					Config: builder.Upsert(egwCp).Upsert(backend).Upsert(virtualCluster).Upsert(staticKey).Upsert(decryptPolicy).Build(),
+					ConfigPlanChecks: resource.ConfigPlanChecks{
+						PreApply: []plancheck.PlanCheck{
+							plancheck.ExpectResourceAction("konnect_event_gateway.tf-test-egw-decrypt", plancheck.ResourceActionCreate),
+							plancheck.ExpectResourceAction("konnect_event_gateway_backend_cluster.tf-test-backend", plancheck.ResourceActionCreate),
```

Comment:
Same here - no need to check for these as already done in specific tests for backend cluster, etc.